manager/integration/tests/test_provisioner.py

```diff
@@ -196,16 +198,44 @@ def test_provisioner_tags(client, core_api, node_default_tags, storage_class, pv
     check_volume_replicas(volumes.data[0], tag_spec, node_default_tags)
 
 
-@pytest.mark.skip(reason="TODO")
-def test_provisioner_fs_format():
+@pytest.mark.parametrize(
+    "inode_size,block_size",
+    [
+        pytest.param("512", "2048"),
+        pytest.param("1024", "1024")
+    ],
+)
+def test_provisioner_fs_format(core_api, storage_class, pvc, pod, inode_size, block_size):  # NOQA
     """
     Context: https://github.com/longhorn/longhorn/issues/4642
     This is to test the FS format options are configured as mentioned in
     the storage class.
 
-    1. Deploy a new storage class 'longhorn-test' with mkfsParams param.
+    1. Deploy a new storage class 'longhorn-provisioner' with mkfsParams param.
     2. Deploy a PVC and POD with the above storage class.
     3. A Longhorn volume should get created, verify the Inode size and
        block size. They should be same as the mkfsParams parameter.
 
     """
+    pod_name = 'provisioner-fs-test'
+    pod['metadata']['name'] = pod_name
+    pod['spec']['volumes'] = [
+        create_pvc_spec(pvc['metadata']['name'])
+    ]
+    pod['spec']['nodeName'] = get_self_host_id()
+    pvc['spec']['storageClassName'] = DEFAULT_STORAGECLASS_NAME
+    storage_class['metadata']['name'] = DEFAULT_STORAGECLASS_NAME
+    storage_class['parameters']['mkfsParams'] = \
+        "-I {} -b {} -O ^metadata_csum,^64bit".format(inode_size, block_size)
+
+    create_storage(core_api, storage_class, pvc)
+    create_and_wait_pod(core_api, pod)
+    pvc_volume_name = get_volume_name(core_api, pvc['metadata']['name'])
+
+    command = "tune2fs -l /dev/longhorn/{} | grep -i 'block size:'"\
+        .format(pvc_volume_name)
+    assert block_size in str(subprocess.check_output(command, shell=True))
```

Comment:
Just to understand, why are we using `shell=True` here?

Reply:
Let pipe command can execute in 1 line command or If I remember correctly it will need using extra codes to fetch out the output of `grep`